public/app/features/variables/state/sharedReducer.ts

```diff
@@ -74,9 +74,18 @@ const sharedReducerSlice = createSlice({
     },
     duplicateVariable: (state: VariablesState, action: PayloadAction<VariablePayload<{ newId: string }>>) => {
       const original = cloneDeep<VariableModel>(state[action.payload.id]);
-      const name = `copy_of_${original.name}`;
-      const newId = action.payload.data?.newId ?? name;
+      let name = `copy_of_${original.name}`;
+      const originalId = action.payload.data?.newId ?? name;
+      let newId = action.payload.data?.newId ?? name;
       const index = getNextVariableIndex(Object.values(state));
+      let idExists = state[newId] != null;
+      let counter = 0;
+      while (idExists && counter < 100) {
```

Comment:
What do you think about making it smarter? I'd try to get the biggest counter and add a new variable after it. Otherwise, if there is an item missing in a list of 20, it can get lost because we're not going to append a new item. 

Also, fixing a 100-item limit doesn't seem to be appropriate since we're adding a limit to avoid an infinite loop. Why don't we just go through the existing IDs, detect the same name ones you're creating, get the counter and add + 1?